### ISSUE TITLE
Can't receive access_token

### ISSUE THREAD

**Issue by AndreyMomot** [raulriera/AuthenticationViewController#6]
Does it work correctly fro Twitter/Facebook or Google+ ?
Can't manage any of them. 
Can anyone provide an example with those networks?
e.g. I have following url for twitter: 
"https://api.twitter.com/oauth/authorize/?client_id=\(clientId)&scope=\(scopes.joined(separator: "+"))&redirect_uri=\(redirectURI)&response_type=code"
accessTokenURL = https://api.twitter.com/oauth/access_token"

redirectURI = "https://www.google.com"

Then, I put my ClientID and Secret, but nothing works. I can login and do not receive my token after switching to redirectURL.

**Comment by raulriera** [comment]
Hi,

The redirect URL needs to be a callback to your app scheme, check the README instructions on how to setup your project https://github.com/raulriera/AuthenticationViewController/blob/master/README.md

**Comment by AndreyMomot** [comment]
Still do not have an answer. Could anyone provide a workable example for Twitter, for example?
Which url should I use for redirectURI and for requestToken there?

**Comment by raulriera** [comment]
I'm the creator of this library, I just said in the previous message what redirectURI to use. 🤔

**Comment by AndreyMomot** [comment]
Yep, thank you. Now I can authenticate, but after that it opens redirect_uri and code from AppDelegate: (authenticationViewController.authenticateWithCode(code)) doesn't retrieve. What am I doing wrong? Doesn't work even with Instagram...

**Comment by raulriera** [comment]
Can you provide some code example?

**Comment by AndreyMomot** [comment]
Here you can find it:
https://github.com/AndreyMomot/AuthenticationViewController

I just can't get what to use for URL Scheme in my case.

**Comment by raulriera** [comment]
```swift
// Before doing this, you should check the url is your redirect-uri before doing anything. Be safe :)
if let components = URLComponents(url: url, resolvingAgainstBaseURL: false), let queryItems = components.queryItems, let code = queryItems.first?.value {
```

Have you checked that the URL you are getting here is the scheme you specified? Does it match the one you used in Instagram?

**Comment by raulriera** [comment]
Make you used `OAuthExample://oauth-callback` in your Instagram app and also that you are getting the same one in the AppDelegate (previous code snippet)
